tests/unit/suites/libraries/joomla/user/JAuthenticationTest.php

```diff
@@ -82,6 +82,7 @@ protected function tearDown()
 	{
 		// Reset the loaded plugins.
 		TestReflection::setValue('JPluginHelper', 'plugins', null);
+		$this->restoreFactoryState();
 
 		parent::tearDown();
 	}
```

Comment:
PHPUnit actually can handle global backup/restoration as long as we haven't turned it off in the config.  I think we should shift toward that API; there are also methods that check for global state changes and raises warnings for it.